libraries/AP_BattMonitor/AP_BattMonitor_GP9301.cpp

```diff
@@ -0,0 +1,87 @@
+#include "AP_BattMonitor_GP9301.h"
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+
+/*
+  "battery" monitor for liquid fuel level systems that give a PWM value indicating quantity of remaining fuel.
+
+  Output is:
+
+    - mAh remaining is fuel level in millilitres
+    - consumed mAh is in consumed millilitres
+    - fixed 1.0v voltage
+ */
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_BattMonitor_GP9301::AP_BattMonitor_GP9301(
```

Comment:
Can we aim for a backend which isn't specific to this chip?  Presumably it's a matter of scaling do we have parameters enough for this?